src/crystal/system/win32/iocp.cr

```diff
@@ -1,32 +1,100 @@
 {% skip_file unless flag?(:win32) %}
 require "c/handleapi"
+require "c/ioapiset"
+require "c/ntdll"
 require "crystal/system/thread_linked_list"
 
 # :nodoc:
-module Crystal::System::IOCP
+struct Crystal::System::IOCP
+  @@wait_completion_packet_methods : Bool? = nil
+
+  {% if flag?(:interpreted) %}
+    # The interpreter doesn't like the interpreted code to dynamically load
+    # symbols from an external library. We thus merely check for their existence
+    # then simply call them, so the interpreter will load/call them properly.
+    def self.wait_completion_packet_methods? : Bool
+      unless (supported = @@wait_completion_packet_methods).nil?
+        return supported
+      end
+      handle = LibC.LoadLibraryExW(Crystal::System.to_wstr("ntdll.dll"), nil, 0)
+      return @@wait_completion_packet_methods = false if handle.null?
+
+      pointer = LibC.GetProcAddress(handle, "NtCreateWaitCompletionPacket")
+      return @@wait_completion_packet_methods = false if pointer.null?
+
+      @@wait_completion_packet_methods = true
+    end
```

Comment:
This could be a `class_getter?`, I guess?


```suggestion
    class_getter? wait_completion_packet_methods : Bool do
      handle = LibC.LoadLibraryExW(Crystal::System.to_wstr("ntdll.dll"), nil, 0)
      return false if handle.null?

      pointer = LibC.GetProcAddress(handle, "NtCreateWaitCompletionPacket")
      return false if pointer.null?

      true
    end
```

Reply:
Probably, but returning from a block is _wrong_ and the getters become confusing with complex initialization, and are also confusing on the truthy vs nil value: will it retry on false?

They're not even thread safe to counterbalance the confusion. Explicit methods are still much better for the time being.

Reply:
> [...] and are also confusing on the truthy vs nil value: will it retry on false?

AFAIR they will retry only on `nil` value, just like in your code.